lib/wprof/gen_prof.rb

```diff
@@ -0,0 +1,62 @@
+class GenProf
+  def initialize(event, rec_type)
+    @event = event
+    @db_runtime = WProf::Config.get_value(:db_runtime)
+    @async_report = WProf::Config.get_value(:async)
+    generate_profiling(rec_type)
+  end
+
+  def generate_profiling(rec_type)
+    common_params
```

Comment:
Rename it to something a little bit more expresive like: `generate_common_params`

Reply:
Yes, is a good idea. Changed.